### PR TITLE
Enable URL-encoded IPv6 addresses to be parsed in JDBC connection strings

### LINKED ISSUE #10: "Hostname" [::1] breaks connection string parsing
`sqlserver://[::1]:1433;database=prisma-demo;user=SA;password=Pr1sm4_Pr1sm4;trustServerCertificate=true;encrypt=true` should be a valid connection string in theory, but:

> Error: Error parsing connection string: Conversion error: Invalid port in `sqlserver://[::1]:1433;database=prisma-demo;user=SA;***;trustServerCertificate=true;encrypt=true`)

See: https://github.com/prisma/prisma-examples/pull/2489/checks?check_run_id=1832964769

### PR DESCRIPTION
**Pull request by yoshuawuyts**
Enables URL-encoded IPv6 addresses to be parsed in JDBC connection strings. Closes #10. Thanks!

### CODE REVIEW

File: src/jdbc.rs [hunk 1]
```diff
@@ -370,6 +402,24 @@ mod test {
         Ok(())
     }
 
+    #[test]
```
Comment (pimeys) [inline src/jdbc.rs:405]: Can we add another test, that tests:

```
jdbc:sqlserver://[::1]:1433;database=prisma-demo;user=SA;password=Pr[sm4_Pr]sm4;trustServerCertificate=true;encrypt=true
```

as the input?

Comment (pimeys) [reply src/jdbc.rs:405]: Also, would this work with the ADO.net strings. Can we use ipv6 addresses in those?

**Review by pimeys — DISMISSED** [review]
see my comment about testing

File: src/jdbc.rs (hunk 1 above)
Comment (yoshuawuyts) [reply src/jdbc.rs:405]: This only addresses the issue for jdbc; I can add support for ado.net too tho

Comment (pimeys) [reply src/jdbc.rs:405]: for completeness sake :)

Comment (yoshuawuyts) [reply src/jdbc.rs:405]: > Can we add another test, that tests:

That's not a valid JDBC input string, so we can't test it unfortunately. It needs to be escaped properly to work, which we're already testing in our escaping tests.

**Comment by yoshuawuyts** [comment]
Actually I'm not sure what an Ado.net IPv6 connection string would look like. It's mostly just a big bag of keys and values, and we currently don't support any known values. @pimeys do you have an example of what you'd like to see parsed here?

File: src/jdbc.rs (hunk 1 above)
Comment (pimeys) [reply src/jdbc.rs:405]: Ok, that's good, so we test 

```
jdbc:sqlserver://[::1]:1433;database=prisma-demo;user=SA;password={Pr[sm4_Pr]sm4};trustServerCertificate=true;encrypt=true
```

already?

**Pull request merged**